controller/instance_manager_controller.go

```diff
@@ -324,6 +327,10 @@ func (imc *InstanceManagerController) syncInstanceManager(key string) (err error
 		if im.Status.CurrentState != types.InstanceManagerStateError {
 			im.Status.CurrentState = types.InstanceManagerStateStopped
 		}
+	} else if pod.DeletionTimestamp != nil {
```

Comment:
can be combinded with above `pod == nil || pod.DeletionTimestamp != nil`

Reply:
The IM object may be just created or the IM pod is just gone when `pod == nil`, then we don't need to update the IM status to `Error`. But there must be something wrong when `pod.DeletionTimestamp != nil` and the IM status should be `Error` in this case. Hence we cannot blindly combine these 2 cases.